servers/dsub/jobs/controllers/aggregation_controller.py

```diff
@@ -24,47 +16,98 @@ def get_job_aggregations(timeFrame, projectId=None):
-            StatusCount(status='Succeeded', count=4),
-            StatusCount(status='Failed', count=6)
-        ]))
-
-    userEntry1 = AggregationEntry(
-        label='circleci',
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=3),
-            StatusCount(status='Failed', count=7)
-        ]))
-
-    userEntry2 = AggregationEntry(
-        label='otherUser',
-        status_counts=StatusCounts(counts=[
-            StatusCount(status='Succeeded', count=7),
-            StatusCount(status='Failed', count=4)
-        ]))
-
-    ownerAggregation = Aggregation(
-        name='AnotherLabel', key='job-id', entries=[entry1, entry2])
-
-    projectAggregation = Aggregation(
-        name='User', key='userId', entries=[userEntry1, userEntry2])
+    window_min = time_convert.time_frame_to_start_time(timeFrame)
+    provider = providers.get_provider(generator.provider_type(), projectId,
+                                      generator.auth_token())
+
+    jobs = generator.generate_jobs_by_window(provider, projectId, window_min)
+
+    total_summary = {}
+    user_summary = {}
+    job_name_summary = {}
+
+    for job in jobs:
+        _count_total_summary(job, total_summary)
+        _count_user_summary(job, user_summary)
+        _count_job_name_summary(job, job_name_summary)
+
+    return _get_stub_data(
+        _to_summary_counts(total_summary),
+        _to_user_aggregation_entry(user_summary),
+        _to_job_name_aggregation_entry(job_name_summary))
+
+
+def _count_total_summary(job, total_summary):
+    if job.status not in total_summary:
+        total_summary[job.status] = 0
+    total_summary[job.status] += 1
+
+
+def _to_summary_counts(summary_counts):
+    return StatusCounts([
+        StatusCount(status, count) for status, count in summary_counts.items()
+    ])
+
+
+def _count_user_summary(job, user_summary):
+    user_id = job.extensions.user_id
+    status = job.status
+
+    user_count = user_summary.get(user_id, {})
+
+    if status not in user_count:
+        user_count[status] = 0
+    user_count[status] += 1
+
+    user_summary[user_id] = user_count
+
+
+def _to_user_aggregation_entry(user_summary):
+    entries = []
+
+    for user_id, countsDict in user_summary.items():
+        countsList = []
+
+        for status, count in countsDict.items():
+            countsList.append(StatusCount(status=status, count=count))
+
+        entries.append(
+            AggregationEntry(
+                label=user_id, status_counts=StatusCounts(countsList)))
+
+    return Aggregation(name='User Id', key='userId', entries=entries)
+
+
+def _count_job_name_summary(job, job_name_summary):
+    job_name = job.name
+    status = job.status
+
+    user_count = job_name_summary.get(job_name, {})
+
+    if status not in user_count:
+        user_count[status] = 0
+    user_count[status] += 1
+
+    job_name_summary[job_name] = user_count
+
+
+def _to_job_name_aggregation_entry(job_name_summary):
```

Comment:
This function and `_to_user_aggregation_entry` can be condensed into one. This is also creating an `Aggregation`, not an `AggregationEntry` so should be named as such. Something like:
```
def _to_aggregation_entry(summary, name, key):
    entries = []
    for key, counts_dict in summary.items():
        counts = [StatusCount(s, c) for s,c in counts_dict.items()]
        entries.append(
            AggregationEntry(
                label=key, status_counts=StatusCounts(counts)))

    return Aggregation(name=name, key=key, entries=entries)
```

Reply:
Nice! Accepted!

Reply:
Sorry I forgot to rename the function, should be `_to_aggregation(...)`